src/util/articleRenderers.ts

```diff
@@ -16,7 +16,7 @@ import {getStrippedTitleFromHtml} from './misc';
 export const renderArticle = async (json: any, articleId: string, dump: Dump, forceParsoidFallback: boolean): Promise<RenderedArticle[]> => {
 
     const articleDetail = await articleDetailXId.get(articleId);
-    const useParsoidFallback = forceParsoidFallback || json.visualeditor?.result;
+    const useParsoidFallback = forceParsoidFallback || json.visualeditor?.result || json?.contentmodel === 'wikitext';
```

Comment:
This line and what is going one in this whole module needs an explanation because I don't get it. Our code never ever should deal with wikitext code. Why we need to "render" anything if we don't have any wikitext?

Reply:
> Our code never ever should deal with wikitext code

However, should we or not, we do get wikitext:
![image](https://user-images.githubusercontent.com/6795530/83947820-13a29d00-a822-11ea-9914-efdb2217a66a.png)
... that was the reason to `renderMCSArticle()` used to fail (there's no particular json properties). As designed `renderMCSArticle()` shouldn't been called in this case - we already have an html.

Moreover, there is the conditional branch in the source code, where we do handle that:
![image](https://user-images.githubusercontent.com/6795530/83947632-e73a5100-a820-11ea-914c-d2c265b6632a.png)

As by me, this is just an one of many things there that we have organized worst. That was 
another reason why I started to refactor the things around in that PR before.

Reply:
Which API call (an example) deliver wikitext?

Reply:
The whole purpose of the module is still not explained.

Reply:
> Which API call (an example) deliver wikitext?
We asking local parsoid for `pagebundle` as following:
```
http://localhost:8000/wiki.archlinux.org/v3/page/pagebundle/Downgrading_packages/606085
```
so it returns both `html` and `data-parsoid` attributes. I'm not sure what we need to ask for a bundle here because, according to the sources, we ignore everything but `html` at that point. Probably we need the urls as following?
```
http://localhost:8000/wiki.archlinux.org/v3/page/html/Downgrading_packages/606085
```

> The whole purpose of the module is still not explained.
Not sure what you mean asking this. There we assemble the content from sections etc.